editor/src/components/inspector/common/inspector-utils.tsx

```diff
@@ -193,3 +196,26 @@ export function getElementsToTarget(paths: Array<ElementPath>): Array<ElementPat
   })
   return result
 }
+
+export function useControlModeWithCycle(
+  initialValue: ControlMode,
+  modes: Array<ControlMode>,
+): [ControlMode | null, React.Dispatch<ControlMode | null>, React.DispatchWithoutAction] {
+  const isCmdPressedRef = useRefEditorState((store) => store.editor.keysPressed.cmd === true)
```

Comment:
instead of this, could you put a flag in cycleToNextMode that goes forward or backwards? this feels like it's mixing abstractions

Reply:
great idea, done! https://github.com/concrete-utopia/utopia/pull/3289/commits/992724e8d61cf404563aa3ac0c0e5ceda497e9d0